odl/discr/discr_mappings.py

```diff
@@ -97,31 +98,30 @@ def __init__(self, map_type, fset, grid, dspace, order='C', linear=False):
                              'to the total number {} of grid points.'
                              ''.format(dspace.size, dspace, grid.ntotal))
 
-        if order not in ('C', 'F'):
+        self._order = str(order).upper()
+        if self._order not in ('C', 'F'):
             raise ValueError('ordering {!r} not understood.'.format(order))
 
-        dom = fset if map_type == 'restriction' else dspace
-        ran = dspace if map_type == 'restriction' else fset
+        dom = fset if map_type_ == 'restriction' else dspace
+        ran = dspace if map_type_ == 'restriction' else fset
         Operator.__init__(self, dom, ran, linear=linear)
         self._grid = grid
-        self._order = order
 
         if self.is_linear:
             if not isinstance(fset, FunctionSpace):
                 raise TypeError('function space {!r} is not a `FunctionSpace` '
                                 'instance.'.format(fset))
-
             if not isinstance(dspace, FnBase):
                 raise TypeError('data space {!r} is not an `FnBase` instance.'
                                 ''.format(dspace))
-
             if fset.field != dspace.field:
-                raise ValueError('field {} of the function space and field'
+                raise ValueError('field {} of the function space and field '
                                  '{} of the data space are not equal.'
                                  ''.format(fset.field, dspace.field))
 
     def __eq__(self, other):
-        return (type(self) == type(other) and
+        return (isinstance(other, type(self)) and
```

Comment:
type equality is essentially instant. Isinstance needs to check all parent classes. Perhaps it was ABCmeta though.